code/modules/clothing/gloves/marine_gloves.dm

```diff
@@ -22,6 +22,42 @@
 	desc = "Insulated marine tactical gloves that protect against electrical shocks."
 	siemens_coefficient = 0
 
+/obj/item/clothing/gloves/marine/corpsman
+	name = "Advanced medical combat gloves"
+	desc = "Advanced medical gloves, these include small electrodes to defibrilate a patiant. No more bulky units!"
+	icon_state = "gloves_marine_medic"
+	var/obj/item/defibrillator/DF
+
+/obj/item/clothing/gloves/marine/corpsman/Initialize()
+	. = ..()
+	DF = new()
+	DF.ready = TRUE
+
+/obj/item/clothing/gloves/marine/corpsman/equipped(mob/user, slot)
+	. = ..()
+	RegisterSignal(user, COMSIG_HUMAN_MELEE_UNARMED_ATTACK, .proc/try_defib)
+
+/obj/item/clothing/gloves/marine/corpsman/unequipped(mob/unequipper, slot)
+	UnregisterSignal(unequipper,COMSIG_HUMAN_MELEE_UNARMED_ATTACK)
+	. = ..()
+
+/obj/item/clothing/gloves/marine/corpsman/examine(mob/user)
+	. = ..()
+	. += DF.maybe_message_recharge_hint(user)
+
+/obj/item/clothing/gloves/marine/corpsman/MouseDrop(atom/over, src_location, over_location, src_control, over_control, params)
+	. = ..()
+	over.MouseDrop_T(DF,usr)
```

Comment:
LIL worried that there is some unintended thing that can some how remove the defib from the gloves but not sure what that would be